src/index.mako.html

```diff
@@ -721,80 +722,64 @@
 
     <script>
       (function() {
+        function getParam(name){
+          return decodeURIComponent((new RegExp('[?|&]' + name + '=' + '([^&;]+?)(&|#|;|$)').exec(location.search)||[,""])[1].replace(/\+/g, '%20'))||null;
+        }
+
+
         // Make sure ajax is using cached requests
         $.ajaxSetup({
           cache: true
         });
-
+ 
+        var adminUrlRegexp = new RegExp('${admin_url_regexp}');
         var module = angular.module('geoadmin');
         var cacheAdd = '${version}' != '' ? '/' + '${version}' : '';
         var pathname = location.pathname.replace(/(index|mobile|embed)\.html$/g, '');
-
-        function getParam(name){
-          return decodeURIComponent((new RegExp('[?|&]' + name + '=' + '([^&;]+?)(&|#|;|$)').exec(location.search)||[,""])[1].replace(/\+/g, '%20'))||null;
+        var staging = 'prod';
+        var testres = /^mf-geoadmin3\.([a-z]+)\.bgdi.ch$/.exec(location.hostname);
+        if (testres && testres.length == 2) {
+          staging = testres[1];
         }
+        var hasOverWrite = false;
 
-        var adminUrlRegexp = new RegExp('${admin_url_regexp}');
-        var wmsUrl = location.protocol + getParam('wms_url');
-        wmsUrl = adminUrlRegexp.test(wmsUrl) ? wmsUrl : null;
-
-        var apiParam = getParam('api_url');
-        var apiEnv = getParam('api_env');
-        var apiUrlBase = '${api_url}';
-
-        if (apiEnv){
-          if (apiEnv == 'prod'){
-            host = '//api3.geo.admin.ch';
-          } else {
-            host = '//mf-chsdi3.' + apiEnv + '.bgdi.ch';
+        function setBackend(prodUrl, techUrl, overWriteParam) {
+          hasOverWrite = false;
+          var url = prodUrl;
+          if (staging != 'prod') {
+            url = techUrl + staging + '.bgdi.ch';
           }
-        } else {
-          host = '${api_url}';
-        }
-
-        if (host.slice(-1) == '/'){
-          host = host.slice(0, -1);
-        }
-
-        if (apiParam){
-          apiUrl = host + '/' + apiParam;
-        } else {
-          apiUrl = host;
+          var overWrite = 'https://' + getParam(overWriteParam);
+          hasOverWrite = adminUrlRegexp.test(overWrite);
+          url = hasOverWrite ? overWrite : url;
+          return url;
         }
 
-        var shopUrl = getParam('shop_url');
-        var shopUrlBase = '${shop_url}';
-
-        if (shopUrl){
-          shopUrl = shopUrl;
-          if (shopUrl == 'dev' || shopUrl == 'int'){
-            shopUrl = '//shop-bgdi.'+ shopUrl + '.bgdi.ch';
-          } else if (shopUrl == 'prod'){
-              shopUrl = '//beta.shop.swisstopo.ch';
-          } else if (shopUrl == 'puzzle'){
-              shopUrl = '//shop-shopbgdi.openshift.puzzle.ch';
-          }
-        } else {
-          shopUrl = shopUrlBase;
-        }
+        // Setting backends
+        var apiUrl = setBackend('https://api3.geo.admin.ch', 'https://mf-chsdi3.', 'api_url');
```

Comment:
Shouldn't we put
`//api3.geo.admin.ch`
`//mf-chsdi3.`

in a config file instead?